test-data/unit/check-unreachable-code.test

```diff
@@ -457,3 +458,44 @@ else:
 reveal_type(x)  # E: Revealed type is 'builtins.str'
 [builtins fixtures/ops.pyi]
 [out]
+
+[case testShortCircuitInExpression]
+import typing
+def f() -> bool: pass
+PY2 = f()
+PY3 = f()
+
+a = PY2 and 's'
```

Comment:
I would add few more tests exchanging the order of ``'s'`` and ``PY2``/``PY3``. So that e.g. ``'s' and False == False``.